internal/controllers/addon/controller.go

```diff
@@ -268,37 +276,14 @@ func (r *AddonReconciler) Reconcile(
 		}
 	}
 
-	// Reconcile Namespace
-	if result, err := r.namespaceReconciler.Reconcile(ctx, addon); err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to reconcile namespace: %w", err)
-	} else if !result.IsZero() {
-		return result, nil
-	}
-
-	// Reconcile addon pullsecrets
-	if result, err := r.secretPropagationReconciler.Reconcile(ctx, addon); err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to ensure pull secret: %w", err)
-	} else if !result.IsZero() {
-		return result, nil
-	}
-
-	// Ensure the creation of the corresponding AddonInstance in .spec.install.olmOwnNamespace/.spec.install.olmAllNamespaces namespace
-	if err := r.ensureAddonInstance(ctx, log, addon); err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to ensure the creation of addoninstance: %w", err)
-	}
-
-	// Reconciler OLM objects
-	if result, err := r.olmReconciler.Reconcile(ctx, addon); err != nil {
-		return ctrl.Result{}, err
-	} else if !result.IsZero() {
-		return result, nil
-	}
-
-	// Reconcile monitoring objects
-	if result, err := r.monitoringFederationReconciler.Reconcile(ctx, addon); err != nil {
-		return ctrl.Result{}, err
-	} else if !result.IsZero() {
-		return result, nil
+	// Run each sub reconciler serially
+	for _, reconciler := range r.subReconcilers {
+		reconcilerType := reflect.TypeOf(reconciler).Elem()
```

Comment:
A way to avoid `reflect` here would be to add a `Name() string` method to the `addonReconciler` interface. Would also make it easier to decouple the display value from the `type` name.